control-plane/pkg/reconciler/kafka/topic.go

```diff
@@ -31,6 +39,61 @@ type TopicConfig struct {
 	BootstrapServers []string
 }
 
+func TopicConfigFromConfigMap(logger *zap.Logger, cm *corev1.ConfigMap) (*TopicConfig, error) {
+	config, err := buildTopicConfigFromConfigMap(cm)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := validateTopicConfig(config); err != nil {
+		return nil, fmt.Errorf("error validating topic config from configmap %s - ConfigMap data: %v", err, cm.Data)
+	}
+
+	logger.Debug("topic config from configmap",
+		zap.Int32("numPartitions", config.TopicDetail.NumPartitions),
+		zap.Int16("replicationFactor", config.TopicDetail.ReplicationFactor),
+		zap.Any("replicationFactor", config.BootstrapServers),
+	)
+
+	return config, nil
+}
+
+func buildTopicConfigFromConfigMap(cm *corev1.ConfigMap) (*TopicConfig, error) {
+	topicDetail := sarama.TopicDetail{}
+
+	var replicationFactor int32
+	var bootstrapServers string
+
+	err := configmap.Parse(cm.Data,
+		configmap.AsInt32(DefaultTopicNumPartitionConfigMapKey, &topicDetail.NumPartitions),
+		configmap.AsInt32(DefaultTopicReplicationFactorConfigMapKey, &replicationFactor),
+		configmap.AsString(BootstrapServersConfigMapKey, &bootstrapServers),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse config map %s/%s: %w", cm.Namespace, cm.Name, err)
+	}
+
+	topicDetail.ReplicationFactor = int16(replicationFactor)
```

Comment:
FYI, I sent this PR https://github.com/knative/pkg/pull/2365 to pkg so that we can drop this hack on `ReplicationFactor`.